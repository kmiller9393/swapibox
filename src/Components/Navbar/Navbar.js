import React, { Component } from 'react';
import PropTypes from 'prop-types';

export default class Navbar extends Component {
  constructor() {
    super();
    this.state = {
      selected: ''
    };
  }

  handleClick = e => {
    e.preventDefault();
    this.setState({ selected: e.target.title });
    this.props.setContainerView(e.target.title);
  };

  render() {
    return (
      <div className="navbar">
        <fieldset>
          <a
            onClick={this.handleClick}
            className={
              this.state.selected === 'people' ? 'glowBtn-active' : 'glowBtn'
            }
            title="people"
          >
            People
          </a>
          <a
            onClick={this.handleClick}
            className={
              this.state.selected === 'planets' ? 'glowBtn-active' : 'glowBtn'
            }
            title="planets"
          >
            Planets
          </a>
          <a
            onClick={this.handleClick}
            className={
              this.state.selected === 'vehicles' ? 'glowBtn-active' : 'glowBtn'
            }
            title="vehicles"
          >
            Vehicles
          </a>
        </fieldset>
      </div>
    );
  }
}

Navbar.propTypes = {
  setContainerView: PropTypes.func
}