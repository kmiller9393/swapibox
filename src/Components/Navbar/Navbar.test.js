import React, { Component } from 'react';
import { shallow } from 'enzyme';
import Navbar from './Navbar';

describe('Navbar', () => {
  let mockSetContainerView;
  let wrapper;
  let mockEvent;

  beforeEach(() => {
    mockEvent = {target: { title: 'people'}, preventDefault: jest.fn()};
    mockSetContainerView = jest.fn();
    wrapper = shallow(<Navbar 
      setContainerView={mockSetContainerView}
        />)
  })

  it('should call handleClick when button is clicked', () => {
    const spy = jest.spyOn(wrapper.instance(), 'handleClick');
    wrapper.instance().forceUpdate();
    wrapper.find('fieldset').childAt(1).simulate('click', mockEvent);
    expect(spy).toHaveBeenCalled();
  });

  it('should match the snapshot', () => {
    expect(wrapper).toMatchSnapshot();
  })
})