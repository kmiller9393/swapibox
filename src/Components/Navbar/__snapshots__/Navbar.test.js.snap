// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Navbar should match the snapshot 1`] = `
ShallowWrapper {
  Symbol(enzyme.__root__): [Circular],
  Symbol(enzyme.__unrendered__): <Navbar
    setContainerView={[Function]}
/>,
  Symbol(enzyme.__renderer__): Object {
    "batchedUpdates": [Function],
    "getNode": [Function],
    "render": [Function],
    "simulateEvent": [Function],
    "unmount": [Function],
  },
  Symbol(enzyme.__node__): Object {
    "instance": null,
    "key": undefined,
    "nodeType": "host",
    "props": Object {
      "children": <fieldset>
        <a
                className="glowBtn"
                onClick={[Function]}
                title="people"
        >
                People
        </a>
        <a
                className="glowBtn"
                onClick={[Function]}
                title="planets"
        >
                Planets
        </a>
        <a
                className="glowBtn"
                onClick={[Function]}
                title="vehicles"
        >
                Vehicles
        </a>
</fieldset>,
      "className": "navbar",
    },
    "ref": null,
    "rendered": Object {
      "instance": null,
      "key": undefined,
      "nodeType": "host",
      "props": Object {
        "children": Array [
          <a
            className="glowBtn"
            onClick={[Function]}
            title="people"
>
            People
</a>,
          <a
            className="glowBtn"
            onClick={[Function]}
            title="planets"
>
            Planets
</a>,
          <a
            className="glowBtn"
            onClick={[Function]}
            title="vehicles"
>
            Vehicles
</a>,
        ],
      },
      "ref": null,
      "rendered": Array [
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "host",
          "props": Object {
            "children": "People",
            "className": "glowBtn",
            "onClick": [Function],
            "title": "people",
          },
          "ref": null,
          "rendered": "People",
          "type": "a",
        },
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "host",
          "props": Object {
            "children": "Planets",
            "className": "glowBtn",
            "onClick": [Function],
            "title": "planets",
          },
          "ref": null,
          "rendered": "Planets",
          "type": "a",
        },
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "host",
          "props": Object {
            "children": "Vehicles",
            "className": "glowBtn",
            "onClick": [Function],
            "title": "vehicles",
          },
          "ref": null,
          "rendered": "Vehicles",
          "type": "a",
        },
      ],
      "type": "fieldset",
    },
    "type": "div",
  },
  Symbol(enzyme.__nodes__): Array [
    Object {
      "instance": null,
      "key": undefined,
      "nodeType": "host",
      "props": Object {
        "children": <fieldset>
          <a
                    className="glowBtn"
                    onClick={[Function]}
                    title="people"
          >
                    People
          </a>
          <a
                    className="glowBtn"
                    onClick={[Function]}
                    title="planets"
          >
                    Planets
          </a>
          <a
                    className="glowBtn"
                    onClick={[Function]}
                    title="vehicles"
          >
                    Vehicles
          </a>
</fieldset>,
        "className": "navbar",
      },
      "ref": null,
      "rendered": Object {
        "instance": null,
        "key": undefined,
        "nodeType": "host",
        "props": Object {
          "children": Array [
            <a
              className="glowBtn"
              onClick={[Function]}
              title="people"
>
              People
</a>,
            <a
              className="glowBtn"
              onClick={[Function]}
              title="planets"
>
              Planets
</a>,
            <a
              className="glowBtn"
              onClick={[Function]}
              title="vehicles"
>
              Vehicles
</a>,
          ],
        },
        "ref": null,
        "rendered": Array [
          Object {
            "instance": null,
            "key": undefined,
            "nodeType": "host",
            "props": Object {
              "children": "People",
              "className": "glowBtn",
              "onClick": [Function],
              "title": "people",
            },
            "ref": null,
            "rendered": "People",
            "type": "a",
          },
          Object {
            "instance": null,
            "key": undefined,
            "nodeType": "host",
            "props": Object {
              "children": "Planets",
              "className": "glowBtn",
              "onClick": [Function],
              "title": "planets",
            },
            "ref": null,
            "rendered": "Planets",
            "type": "a",
          },
          Object {
            "instance": null,
            "key": undefined,
            "nodeType": "host",
            "props": Object {
              "children": "Vehicles",
              "className": "glowBtn",
              "onClick": [Function],
              "title": "vehicles",
            },
            "ref": null,
            "rendered": "Vehicles",
            "type": "a",
          },
        ],
        "type": "fieldset",
      },
      "type": "div",
    },
  ],
  Symbol(enzyme.__options__): Object {
    "adapter": ReactSixteenAdapter {
      "options": Object {
        "enableComponentDidUpdateOnSetState": true,
        "lifecycles": Object {
          "componentDidUpdate": Object {
            "onSetState": true,
          },
          "getSnapshotBeforeUpdate": true,
        },
      },
    },
    "attachTo": undefined,
    "hydrateIn": undefined,
  },
}
`;
